language/move-analyzer/src/symbols.rs

```diff
@@ -191,6 +212,125 @@ pub struct SymbolicatorRunner {
     mtx_cvar: Arc<(Mutex<RunnerState>, Condvar)>,
 }
 
+impl fmt::Display for IdentType {
+    fn fmt(&self, f: &mut fmt::Formatter) -> std::fmt::Result {
+        match self {
+            Self::RegularType(t) => {
+                // Technically, we could use error_format function here to display the "regular"
+                // type, but the original intent of this function is subtly different that we need
+                // (i.e., to be used by compiler error messages) which, for example, results in
+                // verbosity that is not needed here.
+                //
+                // It also seems like a reasonable idea to be able to tune user experience in the
+                // IDE independently on how compiler error messages are generated.
+                write!(f, "{}", type_to_ide_string(t))
+            }
+            Self::FunctionType(mod_ident, name, type_args, args, ret, acquires) => {
+                let type_args_str = if !type_args.is_empty() {
+                    let mut s = "<".to_string();
+                    s.push_str(type_list_to_ide_string(type_args).as_str());
+                    s.push('>');
+                    s
+                } else {
+                    "".to_string()
+                };
+                let acquires_str = if !acquires.is_empty() {
+                    let mut s = " acquires ".to_string();
+                    s.push_str(type_list_to_ide_string(acquires).as_str());
+                    s
+                } else {
+                    "".to_string()
+                };
+                let ret_str = match ret {
+                    sp!(_, Type_::Unit) => "".to_string(),
+                    _ => format!(": {}", type_to_ide_string(ret)),
+                };
+
+                write!(
+                    f,
+                    "fun {}::{}::{}{}({}){}{}",
+                    addr_to_ide_string(&mod_ident.address),
+                    mod_ident.module.value(),
+                    name,
+                    type_args_str,
+                    type_list_to_ide_string(args),
+                    ret_str,
+                    acquires_str
+                )
+            }
+        }
+    }
+}
+
+fn type_to_ide_string(sp!(_, t): &Type) -> String {
+    match t {
+        Type_::Unit => "()".to_string(),
+        Type_::Ref(mut_, s) => format!(
+            "&{} {}",
+            if *mut_ { "mut" } else { "" },
+            type_to_ide_string(s)
+        ),
+        Type_::Param(tp) => {
+            if tp.abilities.is_empty() {
+                format!("{}", tp.user_specified_name)
+            } else {
+                format!(
+                    "{}: {}",
```

Comment:
Do you really want to display these? Feel a bit strange

Reply:
I was wondering about that myself, particularly that we don't do it for struct types (for clarity but also because in some cases we simply don't know). I will remove it for now - we can always add it back if IDE extension users complain.